### PR TITLE
Spark: propagate snapshot properties for RewriteDataFiles and RewritePositionDeleteFiles

### PR DESCRIPTION
**Pull request by advancedxy**
When working with `RewriteDataFiles`, I noticed that the snapshot properties cannot be set via `snapshotProperty(String, String)`.  This PR will fix that issue by propagating snapshot properties into `RewriteDataFilesCommitManager` or `RewritePositionDeletesCommitManager` and all the way down to `RewriteFiles`.

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
@advancedxy: GH actions are down. Hence PR builders are failing.  We need to wait till the status is green in https://www.githubstatus.com/

File: core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java [hunk 1]
```diff
@@ -74,7 +86,9 @@ public void commitFileGroups(Set<RewriteFileGroup> fileGroups) {
     } else {
       rewrite.rewriteFiles(rewrittenDataFiles, addedDataFiles);
     }
-
+    if (!extraCommitSummary.isEmpty()) {
```
Comment (manuzhang) [inline core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java:89]: Do we need this check? If the map is empty, `forEach` won't be executed.

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java [hunk 2]
```diff
@@ -1445,6 +1445,14 @@ public void testRewriteJobOrderFilesDesc() {
     assertThat(actual).as("Number of files order should not be ascending").isNotEqualTo(expected);
   }
 
+  @Test
+  public void testSnapshotProperty() {
+    Table table = createTable(4);
+    Result ignored = basicRewrite(table).snapshotProperty("key", "value").execute();
+    assertThat(table.currentSnapshot().summary())
+        .containsAllEntriesOf(ImmutableMap.of("key", "value"));
```
Comment (manuzhang) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java:1453]: please also check existing summary contents are not lost.

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java [hunk 3]
```diff
@@ -613,6 +614,39 @@ public void testSchemaEvolution() throws Exception {
...
+    Result ignored =
+        SparkActions.get(spark)
+            .rewritePositionDeletes(table)
+            .snapshotProperty("key", "value")
+            .option(SizeBasedFileRewriter.REWRITE_ALL, "true")
+            .execute();
+    assertThat(table.currentSnapshot().summary())
+        .containsAllEntriesOf(ImmutableMap.of("key", "value"));
+
+    // make sure internal produced properties is not lost
```
Comment (ajantha-bhat) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewritePositionDeleteFilesAction.java:636]:
```suggestion
    // make sure internal produced properties are not lost
```

File: spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java [hunk 4]
```diff
@@ -1445,6 +1446,23 @@ public void testRewriteJobOrderFilesDesc() {
...
+    // make sure internal produced properties is not lost
```
Comment (ajantha-bhat) [inline spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java:1455]:
```suggestion
    // make sure internal produced properties are not lost
```

**Review by ajantha-bhat — APPROVED** [review]

**Comment by aokolnychyi** [comment]
I will take a look this week.

**Review by aokolnychyi — APPROVED** [review]

**Comment by aokolnychyi** [comment]
Thanks, @advancedxy! Thanks for reviewing, @manuzhang @ajantha-bhat!

**Pull request merged**